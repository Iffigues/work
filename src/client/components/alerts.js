import React from 'react'
import Alert from '@material-ui/lab/Alert';
import { Context as AlertContext } from "../context/AlertContext";
import { useContext } from 'react';

const Alerts = () => {
  const {
    state: { message, type }
  } = useContext(AlertContext);
  if (message && type) {
  return (
      <Alert severity={type}>{message}</Alert>
      )
  }
  return null;
}

export default Alerts

// use memo
