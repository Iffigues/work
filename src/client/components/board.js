import React from 'react';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import Preview from './preview'
import Game from './game'
import StyledTetrisWrapper from './style/styledBoard'
import StyledTetris from './style/styledBoard'
import { createGame } from '../plugins/createGame';
import { Card, CardContent } from '@material-ui/core';

const boxProps = {
  bgcolor: 'background.paper',
  bordercolor: 'text.primary',
  border: 1,
  borderradius: "borderRadius",
  style: { backgroundColor: 'white', height: '100%', width: '100%'},
  m: 1
};

const Board = ({ mapGame, mapGamePreview, isAlone }) => {
  return (
      <Grid
        container
        direction="row"
        justify="center"
        alignItems="flex-start"
        spacing={3}
        style={{ backgroundColor: '#ffff', width: '100%', height: '100%'}}
      >
        <Grid item xs={8}>
          <Card {...boxProps} variant="outlined">
            <CardContent>
              <Game game={ { game: mapGame, isOtherUser: false } }/>
            </CardContent>
          </Card>
        </Grid>
        <Grid item xs={4}>
          <Preview game={ { game: mapGamePreview, isOtherUser: true } } isAlone={isAlone} />
        </Grid>
      </Grid>
  );
}
export default Board
