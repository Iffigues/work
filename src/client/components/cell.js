import React from 'react';
import StyledCell from './style/styledCell'
import { TETROMINOS } from '../plugins/tetrominos'

const Cell = ({type}) => {
  return (
    <div>
      <StyledCell type={type} color={TETROMINOS[type].color}></StyledCell>
    </div>
  );
}
export default Cell
