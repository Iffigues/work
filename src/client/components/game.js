import React from 'react';
import Cell from './cell'
import StyledStage from './style/styledStage'

const Game = ({game}) => {
  return (
    <div>
      <StyledStage isOtherUser={game.isOtherUser} width={game.game[0].length} height={game.game.length}>
        {game.game.map(row => 
          row.map((cell, x) => 
            <Cell key={x} type={cell}/>
        ))}
      </StyledStage>
    </div>
  );
}
export default Game
