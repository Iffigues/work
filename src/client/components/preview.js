import React from 'react';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import Game from './game'
import { createGame } from '../plugins/createGame'

const boxProps = {
  bgcolor: 'background.paper',
  bordercolor: 'text.primary',
  border: 1,
  borderradius: "borderRadius",
  style: { backgroundColor: 'white', width: '100vw'},
};

const OtherPlayerGrid = ({isAlone, mapGamePreview}) => {
  if (!isAlone) {
    console.log('here', isAlone);
    return (
      <Box mb={3}>
        <Card {...boxProps} style= {{ width: '30vw', height: '65vh' }} variant="outlined">
          <Grid item xs={4}>
            <Game game={ { game: mapGamePreview, isOtherUser: true } } />
          </Grid>
        </Card>
      </Box>
    )
  } else {
    return '';
  }
}

const Preview = ({mapGamePreview, isAlone}) => {
  return (
    <div>
      <Box mb={3}>
        <Card {...boxProps} style= {{ width: '30vw', height: '15vh' }} variant="outlined">
          <CardContent>
            <Typography color="textSecondary" gutterBottom>
            PREVIEW
            </Typography>
          </CardContent>
        </Card>
      </Box>
      <Box mb={3}>
        <Card {...boxProps} style= {{ width: '30vw', height: '5vh' }} variant="outlined">
          <CardContent>
            <Typography color="textSecondary" gutterBottom>
            SCORE
            </Typography>
          </CardContent>
        </Card>
      </Box>
      <OtherPlayerGrid isAlone={isAlone} mapGamePreview={mapGamePreview}/>
    </div>
  );
}
export default Preview
