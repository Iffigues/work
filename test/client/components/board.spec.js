import React from 'react';
import { shallow } from 'enzyme';
import { navBar } from '../../../src/client/components/navBar';

describe('My Test Suite', () => {
  it('My Test Case', () => {
    const wrapper = shallow(<navBar />);
    expect(wrapper).toMatchSnapshot();
  });
});