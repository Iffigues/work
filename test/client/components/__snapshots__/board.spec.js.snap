// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Test Suite My Test Case 1`] = `<navBar />`;
